### PR TITLE
[Parent] Barcode Scanner 3.0

### PR DESCRIPTION
**Pull request by Blackchai**
Behavior unchanged, just had to modify some of the error handling logic and get those tests passing.

### CODE REVIEW

File: apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart [hunk 1]
```diff
@@ -24,24 +24,30 @@ class QRLoginTutorialScreenInteractor {
...
+        case ResultType.Barcode:
+          if (QRUtils.verifySSOLogin(barcodeResult) != null) {
+            result = BarcodeScanResult(true, result: barcodeResult);
+          } else {
+            result = BarcodeScanResult(false, errorType: QRError.invalidQR);
+          }
+          break;
+        case ResultType.Error:
+          result = BarcodeScanResult(false, errorType: QRError.invalidQR);
+          break;
```
Comment (joehoag) [inline apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart:39]: Should the user be able to tell the difference between failed login and ResultType.Error?  Maybe the latter is due to a mechanical failure, lack of camera permissions, blurry scan, etc...?  Should we have different error codes / error messages between the two?

File: apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart [hunk 2]
```diff
@@ -24,24 +24,30 @@ class QRLoginTutorialScreenInteractor {
...
-      if (e.code == BarcodeScanner.CameraAccessDenied) {
+      if (e.code == BarcodeScanner.cameraAccessDenied) {
         result = BarcodeScanResult(false, errorType: QRError.cameraError);
       } else {
         // Unknown error while scanning
         result = BarcodeScanResult(false, errorType: QRError.invalidQR);
       }
-    } on FormatException {
-      // User returned, do nothing
-    } catch (e) {
```
Comment (joehoag) [inline apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart:42]: Do we no longer need a general error catcher?

File: apps/flutter_parent/pubspec.lock [hunk 3]
```diff
@@ -696,6 +696,13 @@ packages:
       url: "https://pub.dartlang.org"
     source: hosted
     version: "3.0.12"
+  protobuf:
+    dependency: transitive
+    description:
+      name: protobuf
+      url: "https://pub.dartlang.org"
+    source: hosted
+    version: "1.0.1"
```
Comment (joehoag) [inline apps/flutter_parent/pubspec.lock:705]: Whoa -- protobufs.  Any indication as to whether this addition beefed up the size of our app?

File: apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart (hunk 2 above)
Comment (Blackchai) [reply apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart:42]: I don't believe we do. The example code from the library changed with the update. I did quite a bit of testing but I can try some more. I wasn't able to trigger this catch.

File: apps/flutter_parent/pubspec.lock (hunk 3 above)
Comment (Blackchai) [reply apps/flutter_parent/pubspec.lock:705]: Not sure, but it was a change introduced by the library, not us. I'm not super stoked on the changes for 3.0 of this library though..

Comment (joehoag) [reply apps/flutter_parent/pubspec.lock:705]: Splunk shows the size of your apk as 64.741 MB, just marginally higher than the normal 64.2 MB.  So it doesn't look like we bloated that much.

File: apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart (hunk 2 above)
Comment (joehoag) [reply apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart:42]: 🤷‍♂️ 👍

File: apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart (hunk 1 above)
Comment (Blackchai) [reply apps/flutter_parent/lib/screens/qr_login/qr_login_tutorial_screen_interactor.dart:39]: This error isn't for a failed login, but only for barcode scanning errors. The login errors are handled further in, at the SplashScreen.

File: apps/flutter_parent/pubspec.lock (hunk 3 above)
Comment (Blackchai) [reply apps/flutter_parent/pubspec.lock:705]: kk thanks!

**Pull request merged**